### PR TITLE
feat(parser): support template literal type

### PR DESCRIPTION
**Pull request by yeonjuan**
## Summary

part of #1244

It's an implementation of parsing template literal type in typescript.

## Test Plan

test cases

### CODE REVIEW

File: internal/ast/js/typescript/TSTemplateElement.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (antonk52) [inline internal/ast/js/typescript/TSTemplateElement.ts:2]: The banner is not required for the new files

File: internal/formatter/builders/js/typescript/TSTemplateElement.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
```
Comment (antonk52) [inline internal/formatter/builders/js/typescript/TSTemplateElement.ts:1]: Here also

**Review by antonk52 — APPROVED** [review]
Great work. Looks like you've beet me to adding this feature, that's awesome!🙂

I know that the Facebook banner is not required for the new files so it can be let go. 

Also nice job with adding the formatter tests too👍

I can see that some tests are failing due to timeout and some did not match on macOS, has this been encountered before? @ematipico

**Comment by yeonjuan** [comment]
> I know that the Facebook banner is not required for the new files so it can be let go.

@antonk52   It was a bad copy-paste. Thanks 👍

**Review by ematipico — APPROVED** [review]

File: internal/formatter/builders/js/typescript/TSTemplateElement.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (ematipico) [inline internal/formatter/builders/js/typescript/TSTemplateElement.ts:2]: Banner to remove

File: internal/js-analysis/evaluators/typescript/TSTemplateElement.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/**
```
Comment (ematipico) [inline internal/js-analysis/evaluators/typescript/TSTemplateElement.ts:1]: Banner to remove

**Review by ematipico — APPROVED** [review]
Great addition to Rome! Let's merge it once the banners are removed

**Comment by yeonjuan** [comment]
@ematipico Oh. my bad. Thanks for the review :) I remove all licenses banner in the PR

**Pull request merged**